### PR TITLE
167 add support to link check

### PR DESCRIPTION
**Pull request by jenkin**
Using [markdown-link-validator](https://github.com/jenkin/markdown-link-validator/tree/feature/sparkfabrik-enhancements) from github repo.

Build the docker image and run validation with `make check`.

> WARNING: this PR makes #171 deprecated.

### CODE REVIEW

**Comment by paolomainardi** [comment]
I am wondering if we should add this to a github actions as well

**Comment by stickgrinder** [comment]
I was asking for a pre-commit hook in the meantime, so no broken links are committed to the repo.

**Comment by jenkin** [comment]
For pre-hooks or github actions, be aware of links to not (yet) existing pages. We have to create them as empty files?

Moreover, some remote links always fail because of bot protection systems (ie. drupal.org), so we need to actively add them to exclusion list (`-i` options)?

Example
```
markdown-link-validator -q -e -o -i ^/downloads/ -i drupal.org -i udemy.com -i sparkfabrik.loc -i acme.sparkfabrik.com ./content
```

If links validation is mandatory for commit or delivery, a clear policy for these cases should be enforced and communicated to developers, imho.

File: custom/package.json [hunk 1]
```diff
@@ -5,12 +5,14 @@
...
+        "start": "node server.js",
+        "check": "markdown-link-validator -q -e -o -i ^/downloads/ -i drupal.org -i udemy.com -i sparkfabrik.loc -i acme.sparkfabrik.com ./content",
+        "test": "echo \"Error: no test specified\" && exit 1"
     },
     "license": "https://creativecommons.org/licenses/by-sa/4.0/",
     "dependencies": {
-      "raneto": "0.17.3",
-      "gulp": "4.0.2"
+        "gulp": "4.0.2",
+        "markdown-link-validator": "github:jenkin/markdown-link-validator#feature/sparkfabrik-enhancements",
```
Comment (paolomainardi) [inline custom/package.json:15]: come mai stiamo usando un fork ? devo aver perso qualche passaggio

Comment (jenkin) [reply custom/package.json:15]: @paolomainardi Ho provato alcune soluzioni già pronte (come il [markdown-link-check](https://github.com/tcort/markdown-link-check) che hai menzionato anche tu o l'originale [markdown-link-validator](https://github.com/webhintio/markdown-link-validator)), ma nessuna soddisfa pienamente i nostri requisiti. Vedi #167 (in particolare la sezione drawbacks).

Alla fine ho optato per [forkare markdown-link-validator](https://github.com/jenkin/markdown-link-validator/tree/feature/sparkfabrik-enhancements) e aggiungere le [funzionalità che ci servono](https://github.com/jenkin/markdown-link-validator/commits/feature/sparkfabrik-enhancements).

Tutte le modifiche fatte sono retrocompatibili per quanto riguarda le API della CLI e a mio avviso di interesse abbastanza generale da poter essere anche rilasciate (ho anche corretto alcune issue ancora aperte). Ma prima di proporre eventuali PR al repo originale aspettavo di integrare il tutto nel playbook e fare più test.

Comment (paolomainardi) [reply custom/package.json:16]: ok :+1:  grazie!

**Comment by stickgrinder** [comment]
This seems ok to me, the bulk of the logic here lies in the markdown link checker, so let's merge this and do some real-world testing.

Thanks @jenkin

**Pull request merged**